src/java/apiview-java-processor/src/main/java/com/azure/tools/apiview/processor/diagnostics/Diagnostics.java

```diff
@@ -1,90 +1,136 @@
 import java.util.List;
-import java.util.Optional;
+import java.util.regex.Pattern;
 
-import static com.azure.tools.apiview.processor.analysers.util.ASTUtils.makeId;
-import static com.azure.tools.apiview.processor.diagnostics.rules.IllegalMethodNamesDiagnosticRule.Rule;
+import static com.azure.tools.apiview.processor.model.Flavor.*;
+
+import static com.azure.tools.apiview.processor.diagnostics.rules.general.IllegalMethodNamesDiagnosticRule.Rule;
 import static com.azure.tools.apiview.processor.diagnostics.rules.RequiredBuilderMethodsDiagnosticRule.DirectSubclassCheckFunction;
 import static com.azure.tools.apiview.processor.diagnostics.rules.RequiredBuilderMethodsDiagnosticRule.ExactTypeNameCheckFunction;
 import static com.azure.tools.apiview.processor.diagnostics.rules.RequiredBuilderMethodsDiagnosticRule.ParameterAllowedTypes;
-import static com.azure.tools.apiview.processor.diagnostics.rules.BadAnnotationDiagnosticRule.BadAnnotation;
-import static com.azure.tools.apiview.processor.model.DiagnosticKind.WARNING;
+import static com.azure.tools.apiview.processor.diagnostics.rules.general.BadAnnotationDiagnosticRule.BadAnnotation;
 
 public class Diagnostics {
-    private final List<DiagnosticRule> diagnostics = new ArrayList<>();
-
-    public Diagnostics() {
-        diagnostics.add(new PackageNameDiagnosticRule());
-        diagnostics.add(new ImportsDiagnosticRule("com.sun"));
-        diagnostics.add(new IllegalPackageAPIExportsDiagnosticRule("implementation", "netty"));
-        diagnostics.add(new NoPublicFieldsDiagnosticRule());
-        diagnostics.add(new UpperCaseNamingDiagnosticRule("URL", "HTTP", "XML", "JSON", "SAS", "CPK", "API"));
-        diagnostics.add(new MissingAnnotationsDiagnosticRule());
-        diagnostics.add(new FluentSetterReturnTypeDiagnosticRule());
-        diagnostics.add(new ConsiderFinalClassDiagnosticRule());
-        diagnostics.add(new IllegalMethodNamesDiagnosticRule(
-            new Rule("Builder$", "tokenCredential"), // it should just be 'credential'
-            new Rule("Builder$", "^set"),            // we shouldn't have setters in the builder
-            new Rule("^isHas"),
-            new Rule("^setHas")
+    final List<DiagnosticRule> rules;
+
+    public Diagnostics(APIListing apiListing) {
+        rules = new ArrayList<>();
+
+        System.out.println("  Setting up diagnostics...");
+
+        Flavor flavor = Flavor.getFlavor(apiListing);
+
+        // Special rules for com.azure or io.clientcore libraries only
+        switch (flavor) {
+            case AZURE: {
+                System.out.println("    Applying com.azure specific diagnostics...");
+                addAzureCoreDiagnostics();
+                break;
+            }
+            case GENERIC: {
+                System.out.println("    Applying io.clientcore specific diagnostics...");
+                addClientCoreDiagnostics();
+                break;
+            }
+            case UNKNOWN:
+            default: {
```

Comment:
For default, should we fallback to using Azure diagnostics to work with older versions of Azure libraries that have already shipped?

Reply:
Sometimes I'm bothered by e.g. Semantic Kernel falling into our Azure SDK diagnostics. I think I would rather we keep it as-is, and make sure all existing libraries are picked up as Azure libraries.